### PR TITLE
fix(location-field): avoid using geocoder generated name if possible

### PR DESCRIPTION
**Pull request by miles-grant-ibigroup**
This PR fixes the bug of duplicate zip codes and states. It turns out the HERE -> Geocode.Earth conversion includes that data in the `name` field which is then being re-appended by the location-field. This is addressed by generating the address name ourselves.

### CODE REVIEW

**Review by philip-cline — APPROVED** [review]

File: packages/location-field/src/utils.ts [hunk 1]
```diff
@@ -21,10 +21,12 @@ const zipOrNeighborhood = ({
 // more layer types with custom rendering.
 const layerDisplayMap = {
   address: (properties: Properties): Label => {
-    const { locality, name, postalcode, region_a: state } = properties;
+    const { housenumber, locality, name, street, postalcode, region_a: state, region } = properties;
```
Comment (binh-dam-ibigroup) [inline packages/location-field/src/utils.ts:24]: Sort props

**Review by binh-dam-ibigroup — APPROVED** [review]
LGTM, it would be nice to see how the house numbers render in storybook.

File: packages/location-field/src/utils.ts (hunk 1 above)
Comment (miles-grant-ibigroup) [reply packages/location-field/src/utils.ts:24]: Sorted in 0ee6e8c85ef16d0a68531803f17a22c393c8f5e2

**Comment by miles-grant-ibigroup** [comment]
The mock autocomplete results are updated in 0ee6e8c85ef16d0a68531803f17a22c393c8f5e2

**Pull request merged**